core/common/src/main/java/alluxio/cli/AbstractCommand.java

```diff
@@ -9,53 +9,38 @@
  * See the NOTICE file distributed with this work for information regarding copyright ownership.
  */
 
-package alluxio.shell.command;
+package alluxio.cli;
 
-import alluxio.client.file.FileSystem;
+import alluxio.exception.ExceptionMessage;
+import alluxio.exception.status.InvalidArgumentException;
 
 import org.apache.commons.cli.CommandLine;
 import org.apache.commons.cli.CommandLineParser;
 import org.apache.commons.cli.DefaultParser;
-import org.apache.commons.cli.Option;
 import org.apache.commons.cli.Options;
 import org.apache.commons.cli.ParseException;
 
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * The base class for all the {@link ShellCommand} classes. It provides a default argument
- * validation method and a place to hold the {@link FileSystem} client.
+ * The base class for all the {@link Command} classes.
  */
 @ThreadSafe
-public abstract class AbstractShellCommand implements ShellCommand {
+public abstract class AbstractCommand implements Command {
 
-  protected FileSystem mFileSystem;
-
-  protected static final String REMOVE_UNCHECKED_OPTION_CHAR = "U";
-  protected static final Option REMOVE_UNCHECKED_OPTION =
-      Option.builder(REMOVE_UNCHECKED_OPTION_CHAR)
-            .required(false)
-            .hasArg(false)
-            .desc("remove directories without checking UFS contents are in sync")
-            .build();
-
-  protected AbstractShellCommand(FileSystem fs) {
-    mFileSystem = fs;
-  }
+  protected AbstractCommand() {}
 
   /**
    * Checks if the arguments are valid.
    *
    * @param args the arguments for the command, excluding the command name and options
-   * @return whether the args are valid
+   * @throws InvalidArgumentException when arguments are not valid
    */
-  protected boolean validateArgs(String... args) {
-    boolean valid = args.length == getNumOfArgs();
-    if (!valid) {
-      System.out.println(getCommandName() + " takes " + getNumOfArgs() + " arguments, " + " not "
-          + args.length + "\n");
+  protected void validateArgs(String... args) throws InvalidArgumentException {
+    if (args.length != getNumOfArgs()) {
```

Comment:
Can we leave `validateArgs` abstract and check `args.length` in `parseAndValidateArgs`? Otherwise subclasses will lose this check when they override `validateArgs`